delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala

```diff
@@ -197,6 +197,39 @@ final class Files(
     } yield res
   }.span("createLink")
 
+  // TODO comments
+  def copy(
+      id: FileId,
+      targetProject: ProjectRef,
+      targetStorageId: Option[IdSegment],
+      tagForCopy: Option[UserTag]
+  )(implicit caller: Caller): IO[FileResource] = {
+    // TODO what if there's an ID clash? Generate one, or let the rejection happen as normal?
+    val newFileId = FileId(id.id.value, targetProject)
+    for {
+      (iri, targetProjectContext) <- newFileId.expandIri(fetchContext.onCreate)
+      (storageRef, storage)       <- fetchActiveStorage(targetStorageId, targetProject, targetProjectContext)
+      file                        <- fetch(id)
+      _                           <- validateStorageTypeForCopy(file.value.storageType, storage)
+      attributes                   = file.value.attributes
```

Comment:
A copy of the physical file must also be done both in the case of local storage and remote storage (that means having to change the storage app too)
We also need a new uuid for this new file